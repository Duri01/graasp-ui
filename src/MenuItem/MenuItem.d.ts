import React from 'react'

export interface MenuItemType {
  id: string
  text?: string
  icon?: React.ReactElement
  path?: string
  key?: string
  onClick?: (path?: string) => void
}

declare const MenuItem: React.FC<MenuItemType>
