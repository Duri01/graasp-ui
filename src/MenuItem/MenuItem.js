import React from 'react';
import MaterialMenuItem from '@material-ui/core/MenuItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';

export const MenuItem = ({ id, text, icon, onClick }) => {
  return (
    <MaterialMenuItem id={id} onClick={onClick} button>
      {icon && <ListItemIcon>{icon}</ListItemIcon>}
      {text && <ListItemText primary={text} />}
    </MaterialMenuItem>
  );
};

export default MenuItem;
