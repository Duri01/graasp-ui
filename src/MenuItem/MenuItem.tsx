import React, { FC, SyntheticEvent, ReactElement } from 'react';
import MaterialMenuItem from '@material-ui/core/MenuItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';

export interface MenuItemProps {
  id?: string;
  text?: string;
  icon?: ReactElement;
  path?: string;
  key?: string;
  onClick?: (event?: SyntheticEvent) => void;
}

export const MenuItem: FC<MenuItemProps> = ({ id, text, icon, onClick }) => {
  return (
    <MaterialMenuItem id={id} onClick={onClick} button>
      {icon && <ListItemIcon>{icon}</ListItemIcon>}
      {text && <ListItemText primary={text} />}
    </MaterialMenuItem>
  );
};

export default MenuItem;
