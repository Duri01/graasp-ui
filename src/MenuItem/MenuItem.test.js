import React from 'react'
import { shallow } from 'enzyme'
import MaterialMenuItem from '@material-ui/core/MenuItem'
import ListItemIcon from '@material-ui/core/ListItemIcon'
import ListItemText from '@material-ui/core/ListItemText'
import ChevronRightIcon from '@material-ui/icons/ChevronRight'
import { MenuItem } from './MenuItem'

const createMenuItemProps = ({ id, text, icon, onClick, key } = {}) => ({
  id,
  text,
  icon,
  onClick,
  key
})

describe('MenuItem', () => {
  let wrapper

  describe('default values', () => {
    beforeAll(() => {
      const props = createMenuItemProps()
      // eslint-disable-next-line react/jsx-props-no-spreading
      wrapper = shallow(<MenuItem {...props} />)
    })

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('renders one <MenuItem /> material component with default values', () => {
      const menuItem = wrapper.find(MaterialMenuItem)
      expect(menuItem).toHaveLength(1)
      const icon = menuItem.find(ListItemIcon)
      expect(icon).toHaveLength(0)
      const text = menuItem.find(ListItemText)
      expect(text).toHaveLength(0)
    })
  })

  describe('prop values', () => {
    const mockFunction = jest.fn()
    beforeAll(() => {
      const props = createMenuItemProps({
        text: 'text',
        id: 'id',
        onClick: mockFunction,
        key: 'key',
        icon: <ChevronRightIcon />
      })
      // eslint-disable-next-line react/jsx-props-no-spreading
      wrapper = shallow(<MenuItem {...props} />)
    })

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('renders one <MenuItem /> material component with default values', () => {
      const menuItem = wrapper.find(MaterialMenuItem)
      expect(menuItem).toHaveLength(1)
      expect(menuItem.prop('id')).toEqual('id')
      expect(menuItem.prop('onClick')).toBe(mockFunction)
      const icon = menuItem.find(ListItemIcon)
      expect(icon).toHaveLength(1)
      const chevron = menuItem.find(ChevronRightIcon)
      expect(chevron).toHaveLength(1)
      const text = menuItem.find(ListItemText)
      expect(text).toHaveLength(1)
      expect(text.prop('primary')).toEqual('text')
    })
  })

  // describe('menu is defined with custom component', () => {
  //   beforeAll(() => {
  //     const props = createMenuItemProps({menu: menuWithCustomMenuItems})
  //     // eslint-disable-next-line react/jsx-props-no-spreading
  //     wrapper = shallow(<MenuItem {...props} />)
  //   })

  //   it('renders correctly', () => {
  //     expect(wrapper).toMatchSnapshot()
  //   })

  //   it('renders one <List /> component with custom menu', () => {
  //     const list = wrapper.find(List)
  //     expect(list).toHaveLength(1)
  //     const items = list.find('div')
  //     expect(items).toHaveLength(menuWithCustomMenuItems.length)
  //   })
  // })
})
