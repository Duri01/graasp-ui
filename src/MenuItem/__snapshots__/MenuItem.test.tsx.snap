// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem default values renders correctly 1`] = `
<WithStyles(ForwardRef(MenuItem))
  button={true}
/>
`;

exports[`MenuItem prop values renders correctly 1`] = `
<WithStyles(ForwardRef(MenuItem))
  button={true}
  id="id"
  onClick={[MockFunction]}
>
  <WithStyles(ForwardRef(ListItemIcon))>
    <ChevronRightIcon />
  </WithStyles(ForwardRef(ListItemIcon))>
  <WithStyles(ForwardRef(ListItemText))
    primary="text"
  />
</WithStyles(ForwardRef(MenuItem))>
`;
